.github/workflows/cont_integration.yml

```diff
@@ -4,141 +4,137 @@ name: CI
-            profile: minimal
-      - name: Rust Cache
-        uses: Swatinem/rust-cache@v2.2.1
-      - name: Pin dependencies for MSRV
-        if: matrix.rust.version == '1.57.0'
-        run: |
-          cargo update -p log --precise "0.4.18"
-          cargo update -p tempfile --precise "3.6.0"
-          cargo update -p rustls:0.21.7 --precise "0.21.1"
-          cargo update -p rustls:0.20.9 --precise "0.20.8"
-          cargo update -p tokio:1.33.0 --precise "1.29.1"
-          cargo update -p tokio-util --precise "0.7.8"
-          cargo update -p flate2:1.0.27 --precise "1.0.26"
-          cargo update -p reqwest --precise "0.11.18"
-          cargo update -p h2 --precise "0.3.20"
-          cargo update -p rustls-webpki:0.100.3 --precise "0.100.1"
-          cargo update -p rustls-webpki:0.101.6 --precise "0.101.1"
-          cargo update -p zip:0.6.6 --precise "0.6.2"
-          cargo update -p time --precise "0.3.13"
-          cargo update -p cc --precise "1.0.81"
-          cargo update -p byteorder --precise "1.4.3"
-          cargo update -p webpki --precise "0.22.2"
-          cargo update -p jobserver --precise "0.1.26"
-      - name: Build
-        run: cargo build ${{ matrix.features }}
-      - name: Test
-        run: cargo test ${{ matrix.features }}
+          fail-mode: true
 
-  check-no-std:
-    name: Check no_std
+  pre-commit-checks:
+    name: "Pre-commit checks: typos, pgp-signed and conventional commits"
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout
-        uses: actions/checkout@v2
-      - name: Install Rust toolchain
-        uses: actions-rs/toolchain@v1
-        with:
-          toolchain: stable
-          override: true
-          profile: minimal
-          # target: "thumbv6m-none-eabi"
-      - name: Rust Cache
-        uses: Swatinem/rust-cache@v2.2.1
-      - name: Check bdk_chain
-        working-directory: ./crates/chain
-        # TODO "--target thumbv6m-none-eabi" should work but currently does not
-        run: cargo check --no-default-features --features bitcoin/no-std,miniscript/no-std,hashbrown
-      - name: Check bdk
-        working-directory: ./crates/bdk
-        # TODO "--target thumbv6m-none-eabi" should work but currently does not
-        run: cargo check --no-default-features --features bitcoin/no-std,miniscript/no-std,bdk_chain/hashbrown
-      - name: Check esplora
-        working-directory: ./crates/esplora
-        # TODO "--target thumbv6m-none-eabi" should work but currently does not
-        run: cargo check --no-default-features --features bitcoin/no-std,miniscript/no-std,bdk_chain/hashbrown
+      - name: checkout
+        uses: actions/checkout@v4
+      - name: Install Nix
+        uses: DeterminateSystems/nix-installer-action@v5
+      # Cache Nix artifacts
+      - uses: DeterminateSystems/magic-nix-cache-action@v2
+      # Commit checks
+      - name: Pre-commit checks
+        run: nix build -L .#ci.pre-commit-check --keep-failed
```

Comment:
**thought:** Oh, interesting. I always run pre-commit checks as `nix develop .#lint -c ./pre-commit-script.sh`, just so it runs exactly like for developers. Need to look into it if it has any advantages this or that way.

Reply:
Yes but then you would need to create a `./pre-commit-script.sh` and maintain it.
Whereas the

```nix
          pre-commit-check = pre-commit-hooks.lib.${system}.run {
            src = ./.;
            hooks = {
              nixpkgs-fmt.enable = true;
              typos.enable = true;
              commitizen.enable = true; # conventional commits
              signedcommits = {
                enable = true;
                name = "signed-commits";
                description = "Check whether the current commit message is signed";
                stages = [ "push" ];
                entry = "${signed-commits}/bin/signed-commits";
                language = "system";
                pass_filenames = false;
              };
            };
          };
        };
```

is easy to maintain since it is in the flake.